src/ClientHandle.cpp

```diff
@@ -1755,6 +1755,28 @@ void cClientHandle::HandleUseEntity(UInt32 a_TargetEntityID, bool a_IsLeftClick)
 
 
 
+void cClientHandle::HandleUseEntityAt(UInt32 a_TargetEntityID, Vector3f a_TargetPos, eHand a_Hand)
+{
+	// If it is a right click, call the entity's OnRightClicked() handler:
+	cWorld * World = m_Player->GetWorld();
+	World->DoWithEntityByID(a_TargetEntityID, [=](cEntity & a_Entity)
+		{
+			if (
+				!cPluginManager::Get()->CallHookPlayerClickingAtEntity(*m_Player, a_Entity, a_TargetPos) &&
+				!m_Player->IsGameModeSpectator()  // Spectators cannot interact with every entity
```

Comment:
This means that the hook is executed for spectators, but then the `OnClickedAt` is not called. This behavior is not documented anywhere in the hook docs, is it even wanted?

Reply:
Not necessary wanted. For me it's totally normal that a plugin must have the priority on all things, even if spectator theoretically can't interact with entity.
Since the behavior is the same for other entity, it's implicitly the same.
Do I answer your question ?

This condition mean: If no plugins interfere, and the player isn't a spectator, then interact